flocker/node/agents/ebs.py

```diff
@@ -650,17 +766,12 @@ def detach_volume(self, blockdevice_id):
             blockdevice_id is not currently 'in-use'.
         """
         ebs_volume = self._get_ebs_volume(blockdevice_id)
-        volume = _blockdevicevolume_from_ebs_volume(ebs_volume)
-        if (volume.attached_to is None or
-                ebs_volume.status != 'in-use'):
+        if ebs_volume.status != 'in-use':
```

Comment:
Yes, this change fixes one the two intermittent failing tests (FLOC-2540 and FLOC-2639).